server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -1925,8 +1925,20 @@ public RefreshResult refresh(String source) throws EngineException {
     }
 
     @Override
-    public RefreshResult maybeRefresh(String source) throws EngineException {
-        return refresh(source, SearcherScope.EXTERNAL, false);
+    public void maybeRefresh(String source, ActionListener<RefreshResult> listener) {
+        externalRefresh(source, false, listener);
+    }
+
+    @Override
+    public void externalRefresh(String source, ActionListener<RefreshResult> listener) {
+        externalRefresh(source, true, listener);
+    }
+
+    protected void externalRefresh(String source, boolean block, ActionListener<Engine.RefreshResult> listener) {
```

Comment:
What's the relation between this and the same-named function in stateless? Should stateless override this as well?

Reply:
I think there is a misunderstanding. I was just explaining the `externalRefresh`. The serverless part is already there since that was a prerequisite for throttling! I think i didn't explain it clearly. IndexEngine overrides this, and this `externalRefresh` is method of Engine. We do not need to override it here. This is the bit that is not needed in this PR.